src/measure_theory/integral/bochner.lean

```diff
@@ -1005,6 +1005,16 @@ begin
   rw [← lt_top_iff_ne_top], convert hfi.has_finite_integral, ext1 x, rw [nnreal.nnnorm_eq]
 end
 
+lemma of_real_integral_eq_lintegral_of_real {f : α → ℝ} (hfi : integrable f μ) (f_nn : 0 ≤ f) :
+  ennreal.of_real (∫ x, f x ∂μ) = ∫⁻ x, ennreal.of_real (f x) ∂μ :=
+begin
+  set g := real.to_nnreal ∘ f with def_g,
+  have coe_g : (λ x, (g x : ℝ)) = f, from real.coe_comp_to_nnreal_comp f_nn,
+  have of_real_g : (λ x, (g x : ℝ≥0∞)) = (λ x, ennreal.of_real (f x)),
+  from ennreal_coe_comp_to_nnreal_comp f_nn,
+  rw [←coe_g, ←lintegral_coe_eq_integral g (by rwa ←coe_g at hfi), of_real_g, coe_g],
+end
```

Comment:
Since we already have the same statement for norms of functions (`of_real_integral_norm_eq_lintegral_nnnorm`), we could use it here and simply prove that `∥f x∥ = f x` since `f x` is nonnegative. 
```suggestion
calc ennreal.of_real (∫ x, f x ∂μ) = ennreal.of_real ∫ x, ∥f x∥ ∂μ :
  by { congr, ext1 x, rw [real.norm_eq_abs, abs_eq_self.mpr (f_nn x)], }
... = ∫⁻ x, ∥f x∥₊ ∂μ : of_real_integral_norm_eq_lintegral_nnnorm hfi
... = ∫⁻ x, ennreal.of_real (f x) ∂μ :
  by { congr, ext1 x, rw [← of_real_norm_eq_coe_nnnorm, real.norm_eq_abs,
                          abs_eq_self.mpr (f_nn x)] }
```
This proof also avoids the two other new lemmas about compositions of coe and to_nnreal. I think we should drop those two, since they are close to existing lemmas like `real.coe_to_nnreal`.

Reply:
I agree with @RemyDegenne that we should drop the two lemmas. The other lemma is close to `ennreal.of_real_eq_coe_nnreal`.
Here is an alternate shorter proof
```suggestion
by simp_rw [of_real_eq_coe_nnreal (f_nn _), lintegral_coe_eq_integral (λ x, ⟨f x, f_nn x⟩) hfi,
    nnreal.coe_mk]
```

Reply:
These sorts of golfs are exactly what I was hoping for when I [asked on Zulip](https://leanprover.zulipchat.com/#narrow/stream/217875-Is-there-code-for-X.3F/topic/lintegral_coe_eq_coe_integral/near/280088881).

Thank you both for simplifying this a lot!

Reply:
That thread did lead me here, so it served its purpose :-)